### PR TITLE
[docs] add openapi spec to docs

### PR DESCRIPTION
**Pull request by yanxi0830**
# What does this PR do?
- modify openapi generator to add coming soon tag for unimplemented api
- sphinx-redocs extension for openapi spec to readthedocs page

## Test Plan

https://github.com/user-attachments/assets/b4c7eebc-2361-4198-a987-dbfbcff914cf

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Ran pre-commit to handle lint / formatting issues.
- [ ] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [ ] Wrote necessary unit or integration tests.

### CODE REVIEW

File: docs/source/index.md [hunk 1]
```diff
@@ -94,5 +94,7 @@ getting_started/index
 concepts/index
 distributions/index
 contributing/index
+distribution_dev/index
```
Comment (ashwinb) [inline docs/source/index.md:97]: this should go away. I deleted and moved the content inside another section.

File: docs/source/index.md [hunk 2]
```diff
@@ -94,5 +94,7 @@ getting_started/index
...
 references/index
+api_reference/index
```
Comment (ashwinb) [inline docs/source/index.md:99]: can we put this inside `References` ?

File: docs/openapi_generator/pyopenapi/generator.py [hunk 3]
```diff
@@ -438,6 +438,14 @@ def _build_extra_tag_groups(
         return extra_tags
 
     def _build_operation(self, op: EndpointOperation) -> Operation:
+        if op.defining_class.__name__ in [
+            "SyntheticDataGeneration",
+            "PostTraining",
+            "BatchInference",
+        ]:
+            op.defining_class.__name__ = f"{op.defining_class.__name__} (Coming Soon)"
```
Comment (ashwinb) [inline docs/openapi_generator/pyopenapi/generator.py:446]: 🤟

Comment (ashwinb) [reply docs/openapi_generator/pyopenapi/generator.py:446]: I like that you are starting to hack the openapi generator too hahaha

File: docs/resources/llama-stack-spec.html [hunk 4]
```diff
@@ -19,9 +19,9 @@
             spec = {
     "openapi": "3.1.0",
     "info": {
-        "title": "[DRAFT] Llama Stack Specification",
+        "title": "Llama Stack Specification",
         "version": "alpha",
-        "description": "This is the specification of the llama stack that provides\n                a set of endpoints and their corresponding interfaces that are tailored to\n                best leverage Llama Models. The specification is still in draft and subject to change.\n                Generated at 2024-11-19 09:14:01.145131"
+        "description": "This is the specification of the Llama Stack that provides\n                a set of endpoints and their corresponding interfaces that are tailored to\n                best leverage Llama Models. Generated at 2024-11-22 17:23:55.034164"
```
Comment (ashwinb) [inline docs/resources/llama-stack-spec.html:24]: This `Generated at` is not useful either and random spurious change happens. We should kill it. Git shows us the commit anyway.

**Comment by ashwinb** [comment]
Looking at the video, when you click on the API reference, do you end up transitioning to a completely different template (i.e., the main documentation nav, etc. is lost?) Just curious

File: docs/source/index.md (hunk 1 above)
Comment (yanxi0830) [reply docs/source/index.md:97]: done!

File: docs/source/index.md (hunk 2 above)
Comment (yanxi0830) [reply docs/source/index.md:99]: done

**Comment by yanxi0830** [comment]
> Looking at the video, when you click on the API reference, do you end up transitioning to a completely different template (i.e., the main documentation nav, etc. is lost?) Just curious

Yeah, the main nav bar will be gone with rendering with Redocly. The sphinx-openapi extension makes it look native, but less interactive & informative. 

image

This line defines the theme, so we could edit it to match the main doc's theme I think. 
```
redoc_uri = "https://cdn.jsdelivr.net/npm/redoc@next/bundles/redoc.standalone.js"
```

**Comment by ashwinb** [comment]
Rendering locally and testing a bit

**Review by ashwinb — APPROVED** [review]
I think stylistically we will likely need some changes here, but this is a major improvement and allows access to API reference directly. Let's get this in!

**Comment by ashwinb** [comment]
I think some simple font / line-height / main text color overrides should make it feel a lot closer. And then we can give the left nav the same gradient too.

**Pull request merged**